### ISSUE TITLE
Non PSR-4 standardized files in autoload dir

### ISSUE THREAD

**Issue by windaishi** [composer/composer#9781]
Hi folks,

we have the following file structure in our project:

```
src/
 | - SomeClass.php
 | - SomeClass.unit.php
```

So for each class file there is a Unit-Test file directly next to the original class file.

When I execute the following command:
```
composer install --optimize-autoloader
```

I get the following output:
```
Class Project\SomeClassTest located in ./src/SomeClass.unit.php does not comply with psr-4 autoloading standard. Skipping.
```

Since our project has pretty much of those Test-Classes and files, the console output is spammed with those messages.

Can I try to hide this message for those files? Actually I don't want them to end up in the auto-loader. Is there a way to ignore the files?

I already tried the following in my composer.json:
```json
    "autoload": {
        "psr-4": {
            "Project\\": "src/"
        },
        "exclude-from-classmap": [
            "**.unit.php",
        ]
    },
```
but it did not help.

Should it work like this and is it a bug? Is this "not supported"? Am I doing it wrong?

I would be very thankful to get some help from you.

**Comment by stof** [comment]
it should be `**/*.unit.php` instead, I think

**Comment by windaishi** [comment]
Thanks @stof 

Could solve it with your help.

Actually you were my debug rubber duck :D. My origin solution did work, but it did not recognized it because there were still a ton of warnings when dumping the autoloader. But when I tried out your solution I could find, that they were all from dependencies of the package. So the warnings actually did disappear, but only for the package itself.
After I added the exclude rule to all packages, the warnings were gone forever. Thank you for helping me :-P

For someone who might stumble across this:
Both solutions work: ```**/*.unit.php``` and ```**.unit.php```
